src/game/shell.c

```diff
@@ -182,6 +187,78 @@ void Shell_Wait(int nticks)
     }
 }
 
+void Shell_GetScreenshotName(char *str)
+{
+    // Get level title of unknown length
+    char level_title[100];
+    sprintf(level_title, "%s", g_GameFlow.levels[g_CurrentLevel].level_title);
+
+    // Trim level titles that are too long
+    if (strlen(level_title) >= LEVEL_TITLE_SIZE) {
+        level_title[LEVEL_TITLE_SIZE] = '\0';
+    }
+
+    // Prepare level title for screenshot
+    char *check = level_title;
+    bool prev_us = true; // '_' after timestamp before title
+    int idx = 0;
+
+    while (*check != '\0') {
+        if ((*check < 'A' || *check > 'Z') && (*check < 'a' || *check > 'z')
+            && (*check < '0' || *check > '9') && *check != '\'' && *check != '.'
+            && *check != ' ') {
+            // Strip non alphanumeric chars
+            memmove(
+                &level_title[idx], &level_title[idx + 1],
+                strlen(level_title) - idx);
+        } else if (*check == ' ') {
+            // Replace spaces with a single underscore
+            if (prev_us) {
+                memmove(
+                    &level_title[idx], &level_title[idx + 1],
+                    strlen(level_title) - idx);
+            } else {
+                *check++ = '_';
+                idx++;
+                prev_us = true;
+            }
+        } else if (*check == '\'' || *check == '.') {
+            // Strip ' and .
+            memmove(
+                &level_title[idx], &level_title[idx + 1],
+                strlen(level_title) - idx);
```

Comment:
I think this condition can be merged with the first one, simplifying the first condition to
```c
        if ((*check < 'A' || *check > 'Z') && (*check < 'a' || *check > 'z')
            && (*check < '0' || *check > '9') && *check != ' ') {
```

Reply:
A - Z a-z 0-9 can be replaced with `iaslnum()` because you also have to handle Extended ASCII such as á

Reply:
AFAIK sprite numbers for characters such as á in TR1 do not conform to any known encoding.

Reply:
I have no idea on language support other than English if I'm being honest. Just let me know if it needs changed or not.

Reply:
from Arsunt 
![image](https://user-images.githubusercontent.com/7146763/148561667-d2a79aff-e754-4b1e-aa2d-12c584c5df24.png)
which is most definitely not a standard encoding, but probably something that needs to be handled, to convert the encoding to EASCII/UTF16 for the filename.

Reply:
The comment about merging second and third branch still stands.

Reply:
Ok so do we need to open an issue for that?

Also oof I see what you mean now. When I removed the validate gameflow function and moved this here, I didn't clean it up properly. I think it's good now?

Reply:
Regarding the encoding, if you want… I don't think I'll ever want to touch this part myself.

Reply:
Oh wait... I just realized I don't think I need `|| *check == '\'' || *check == '.'` at all anymore. Before they were allowed in the gameflow level titles but not in the screenshot name. Now I just need to strip. I'll push.